### PR TITLE
Remove old research age checks

### PR DESCRIPTION
**Pull request by denravonska**
Remove checks which are never triggered due to the grandfather height. The rationale is to get rid of old, unused code to be able to make further improvements.

I have synced from 0 twice using this but I will do another run tonight just to ensure that it works.

### CODE REVIEW

File: src/kernel.cpp [hunk 1]
```diff
@@ -312,7 +304,7 @@ int64_t GetRSAWeightByCPID(std::string cpid)
                         {
                                 weight = (UntrustedHost.owed*14) + UntrustedHost.Magnitude;
                                 if (fTestNet && weight < 0) weight = 0;
```
Comment (denravonska) [inline src/kernel.cpp:306]: Remove this line. Now it's the same for testnet and prod.

File: src/kernel.cpp [hunk 2]
```diff
@@ -379,7 +371,7 @@ int64_t GetRSAWeightByBlock(MiningCPID boincblock)
     }
 
     if (fTestNet && rsa_weight < 0) rsa_weight = 0;
```
Comment (denravonska) [inline src/kernel.cpp:373]: Remove this line.

**Comment by jamescowens** [comment]
Let's get a move on this. Reviewers please comment. It looks like it has to be rebased.

**Comment by cyrossignol** [comment]
I can pick this up, since I'm contributing to a bunch of those conflicts 🙂

**Pull request closed without merging**